v0/util/cache.js

```diff
@@ -17,12 +17,22 @@ class Cache {
 
     return storeFunction().then(result => {
       // store in cache if the value is valid, else skip
+      let retVal = result;
       if (result) {
-        this.cache.set(key, result);
+        if (typeof result === "object" && result.value && result.age) {
```

Comment:
This is incorrect. It'll fail for cases where `value` is falsy (`false`, `0`, "", etc.)